erlang/blocktree/blocktree.erl

```diff
@@ -258,3 +264,22 @@ get_block_by_id(Id, TreeData)
     ?assertMatch({ok, _}, Result, "cannot find block by id"),
     {ok, Block} = Result,
     Block.
+
+get_all_longest_branches(TreeData) 
+  when is_record(TreeData, tree_data) ->
+    #tree_data{block_map = BlockMap} = TreeData,
+    MaxHtFn = fun(_, V, Max) -> max(maps:get(height, V), Max) end,
+    MaxHt = maps:fold(MaxHtFn, 0, BlockMap),
+    
+    HtExtractFn = fun(_, V, List) -> 
+			  Ht = maps:get(height, V),
+			  if Ht == MaxHt ->
+				  [V | List];
```

Comment:
If there are 2 chains with height MaxHt, they will get appended to each other D:

Reply:
Oh wait, that's why it's get_all_longest_branches :D

Reply:
I think you could also just do maps:values(maps:filter(fun(_, V, List) -> maps:get(height, V) == MaxHt end, BlockMap)) instead of fancy fold :o

Reply:
sweet fold still!

Reply:
I mean, I learned like one cool functional language function, that is all I need :P

Reply:
You could push a commit to this branch fixing it O:

Reply:
:o